src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -106,15 +111,23 @@ private Task ProcessAuthenticationAsync(bool isAsync = false, CancellationToken
         {
             ThrowIfExceptional();
 
-            if (NetSecurityTelemetry.Log.IsEnabled())
+            Activity? activity = s_activitySource.StartActivity(ActivityName, IsServer ? ActivityKind.Server : ActivityKind.Client);
+            try
             {
-                return ProcessAuthenticationWithTelemetryAsync(isAsync, cancellationToken);
+                if (NetSecurityTelemetry.Log.IsEnabled())
+                {
+                    return ProcessAuthenticationWithTelemetryAsync(isAsync, cancellationToken);
+                }
+                else
+                {
+                    return isAsync ?
+                        ForceAuthenticationAsync<AsyncReadWriteAdapter>(IsServer, null, cancellationToken) :
+                        ForceAuthenticationAsync<SyncReadWriteAdapter>(IsServer, null, cancellationToken);
+                }
             }
-            else
+            finally
             {
-                return isAsync ?
-                    ForceAuthenticationAsync<AsyncReadWriteAdapter>(IsServer, null, cancellationToken) :
-                    ForceAuthenticationAsync<SyncReadWriteAdapter>(IsServer, null, cancellationToken);
+                activity?.Stop();
```

Comment:
Won't this stop the activity possibly before authentication has completed? There's no await above.